parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileWriter.java

```diff
@@ -1317,6 +1324,27 @@ private int toIntWithCheck(long size) {
     return (int)size;
   }
 
+  private void mergeColumnStatistics(Statistics<?> statistics) {
+    if (!currentStatisticsAreValid) {
```

Comment:
Are we able to reuse `currentStatistics.isEmpty()` in this case? We can simply set currentStatistics to empty if any unexpected statistics are to merge.

Reply:
Do we support the case where the input statistics are empty? In that case, the `currentStatistics` will be empty after merging the first statistics. Then we have no idea whether the statistics are valid or not.

Reply:
From the comment below, I think it serves the same purpose to mark a statistics as invalid. Merging an empty stats should invalidate aggregate statistics and ColumnIndexBuilder, IMO.

```
  /**
   * Returns a boolean specifying if the Statistics object is empty,
   * i.e does not contain valid statistics for the page/column yet
   * @return true if object is empty, false otherwise
   */
  public boolean isEmpty() {
    return !hasNonNullValue && !isNumNullsSet();
  }

```

Reply:
OK, I updated the implementation.